Shopify/cli
# Use TS uploader for shopify theme share command

### WHY are these changes introduced?

Fixes https://github.com/Shopify/develop-advanced-edits/issues/25

The `shopify theme share` command was still calling the Ruby implementation and we wanted it to call the Typescript implementation like the other commands.

### WHAT is this pull request doing?

This PR switches `run` to call the TS uploader.
I have also made a `legacyRun` command that still uses the Ruby version as a fallback. We can remove the fallback in a later PR.

### How to test your changes?

Pull down the branch
Build the branch
run the `shopify theme share` command and check that the output looks the same as before.


### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
